arm-batch/2015-12-01/BatchManagement.json

```diff
@@ -0,0 +1,1340 @@
+            "description": "The name of the Batch account."
+          },
+          {
+            "name": "maxresults",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "Gets or sets the maximum number of items to return in the response."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/ListApplicationsResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Batch/locations/{locationName}/quotas": {
+      "get": {
+        "tags": [
+          "Subscription"
+        ],
+        "operationId": "Subscription_GetSubscriptionQuotas",
+        "description": "The Get Subscription Quotas operation returns the quotas of the subscription in the Batch service.",
+        "parameters": [
+          {
+            "name": "locationName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The region of the quotas belongs."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/SubscriptionQuotasGetResult"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AutoStorageBaseProperties": {
+      "properties": {
+        "storageAccountId": {
+          "type": "string",
+          "description": "Gets or sets the resource id of the storage account to be used for auto storage."
+        }
+      },
+      "required": [ "storageAccountId" ],
+      "description": "Gets or sets properties relating to auto storage."
+    },
+    "AccountBaseProperties": {
+      "properties": {
+        "autoStorage": {
+          "$ref": "#/definitions/AutoStorageBaseProperties",
+          "description": "Gets or sets properties relating to auto storage."
+        }
+      },
+      "description": "The properties of a Batch account."
+    },
+    "BatchAccountCreateParameters": {
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "The region in which the account is created."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The user specified tags associated with the account."
+        },
+        "properties": {
```

Comment:
Please take a look at the extension [`x-ms-client-flatten`](https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-client-flatten). By applying this extension over here and on autoStorage property of AccountBaseProperties you can bring storageAccountId property all the way upto here.
This is much easier for the user
```javascript
var batchAccountCreateParameters = {
  location: "West US",
  tags: {tag1: value1},
  storageAccountId: "testaccount123"
}
```
than this:
```javascript
var batchAccountCreateParameters = {
  location: "West US",
  tags: {tag1: value1},
  properties: {
    autoStorage: {
      storageAccountId: "testaccount123"
    }
  }
}
```